File: DSA/Arrays/Code/5_ArrayInsertion.cs
using System;
using System.Collections.Generic;

// Array Insertion in C#
// Insert an element at a specific position

class ArrayInsertion {
    
    // Function to insert element in array
    static void InsertElement(int[] arr, int n, int position, int value) {
        // Shift elements to the right
        for (int i = n - 1; i >= position; i--) {
            arr[i + 1] = arr[i];
        }
        // Insert new element
        arr[position] = value;
    }
    
    static void DisplayArray(int[] arr, int n) {
        for (int i = 0; i < n; i++) {
            Console.Write(arr[i] + " ");
        }
        Console.WriteLine();
    }
    
    static void Main() {
        // Method 1: Using array (fixed size)
        int[] arr = new int[10];
        int n = 5;
        arr[0] = 10; arr[1] = 20; arr[2] = 30; arr[3] = 40; arr[4] = 50;
        
        Console.Write("Original array: ");
        DisplayArray(arr, n);
        
        // Insert at beginning
        InsertElement(arr, n, 0, 5);
        n++;
        Console.Write("After inserting 5 at beginning: ");
        DisplayArray(arr, n);
        
        // Method 2: Using List<T> (dynamic size)
        List<int> list = new List<int>();
        list.Add(10);
        list.Add(20);
        list.Add(30);
        list.Add(40);
        list.Add(50);
        
        Console.WriteLine("Original List: " + string.Join(" ", list));
        
        // Insert at position
        list.Insert(2, 25);
        Console.WriteLine("After inserting 25 at position 2: " + string.Join(" ", list));
        
        // Insert at beginning
        list.Insert(0, 5);
        Console.WriteLine("After inserting 5 at beginning: " + string.Join(" ", list));
    }
}

File: DSA/Arrays/Code/5_ArrayInsertion.cpp
#include <iostream>
using namespace std;

// Array Insertion in C++
// Insert an element at a specific position in an array

void insertElement(int arr[], int &n, int position, int value) {
    // Shift elements to the right
    for (int i = n - 1; i >= position; i--) {
        arr[i + 1] = arr[i];
    }
    // Insert new element
    arr[position] = value;
    n++;  // Increase size
}

void displayArray(int arr[], int n) {
    for (int i = 0; i < n; i++) {
        cout << arr[i] << " ";
    }
    cout << "\n";
}

int main() {
    int arr[10] = {10, 20, 30, 40, 50};
    int n = 5;
    
    cout << "Original array: ";
    displayArray(arr, n);
    
    // Insert at beginning
    insertElement(arr, n, 0, 5);
    cout << "After inserting 5 at beginning: ";
    displayArray(arr, n);
    
    // Insert at middle
    insertElement(arr, n, 3, 25);
    cout << "After inserting 25 at position 3: ";
    displayArray(arr, n);
    
    // Insert at end
    insertElement(arr, n, n, 60);
    cout << "After inserting 60 at end: ";
    displayArray(arr, n);
    
    return 0;
}
